tpu-client/src/quic_client.rs

```diff
@@ -15,11 +15,59 @@ use {
     },
     lazy_static::lazy_static,
     solana_sdk::transport::Result as TransportResult,
-    std::{net::SocketAddr, sync::Arc},
+    std::{
+        net::SocketAddr,
+        sync::{Arc, Condvar, Mutex, MutexGuard},
```

Comment:
any reason to choose these vs their async analogs?

Reply:
Yes. We need the sync one before spawning into the runtime.

Reply:
ah i see, we're outside the async context. :+1: